src/themes.lisp

```diff
@@ -26,9 +26,25 @@ function that takes a DOCUMENT and returns NIL or a STRING for template insertio
   (or (find-package (format nil "~:@(coleslaw.theme.~A~)" name))
       (error 'theme-does-not-exist :theme name)))
 
-(defun theme-fn (name &optional (package (theme *config*)))
+(defun get-djula-theme (name)
+  (symbol-value (intern (string-upcase (format nil "*djula-~A-template*" (string name))))))
+
+(defun theme-fn (name &optional
+                        (package (theme *config*))
+                        (templating-engine *templating-engine*))
   "Find the symbol NAME inside PACKAGE which defaults to the theme package."
-  (find-symbol (princ-to-string name) (theme-package package)))
+  (case templating-engine
+    (cl-closure (let ((func (find-symbol (princ-to-string name)
+                                         (theme-package package))))
+                  (lambda (&rest rest)
+                    (funcall func rest))))
+    (djula (lambda (&rest rest)
+             (with-output-to-string (stream)
+               (apply #'render-template*
+                      (get-djula-theme name)
+                      stream
+                      rest))))
+    (otherwise (error "Unkown templating engine found"))))
```

Comment:
For ease of debuggability, we should include the value that we failed to match, templating-engine in this case.